lib/src/equality.dart

```diff
@@ -202,14 +202,14 @@ class ListEquality<E> implements Equality<List<E>> {
   bool isValidKey(Object? o) => o is List<E>;
 }
 
-abstract class _UnorderedEquality<E, T extends Iterable<E>?>
+abstract class _UnorderedEquality<E, T extends Iterable<E>>
     implements Equality<T> {
   final Equality<E> _elementEquality;
 
   const _UnorderedEquality(this._elementEquality);
 
   @override
-  bool equals(T elements1, T elements2) {
+  bool equals(T? elements1, T? elements2) {
```

Comment:
Adding the `?` here does fix the backwards compatibility issue.
I guess it's sufficient as long as you don't cast to `Equality<Set<E>>`, and if you did, it wouldn't work anyway.

I still want to remove this `?` too, but that'll likely require a major version bump.

Reply:
> I still want to remove this `?` too, but that'll likely require a major version bump.

Yeah I think that would have been a better migration. We can keep #214 open and tag it for the next major version release.

> We're also inconsistent in that other equalities have not been extended to nullable types).

`IterableEquality`, `MapEquality`, and `ListEquality` all have this same pattern, they accept nullable arguments but the declared `Equality` interface is with a non-nullable type.

Reply:
Ah yes. We are consistently weird :)

LGTM!